src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -1766,7 +1765,8 @@ static bool clear_ic_at_addr(CompiledMethod* caller_nm, address call_addr, bool
 // sites, and static call sites. Typically used to change a call sites
 // destination from compiled to interpreted.
 //
-methodHandle SharedRuntime::reresolve_call_site(JavaThread *thread, TRAPS) {
+methodHandle SharedRuntime::reresolve_call_site(TRAPS) {
+  JavaThread* thread = THREAD->as_Java_thread();
   ResourceMark rm(thread);
   RegisterMap reg_map(thread, false);
   frame stub_frame = thread->last_frame();
```

Comment:
The call to thread->last_frame() may require JavaThread.  I think it's a good change to remove the thread parameter.